### PR TITLE
Edit duedate tff154

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary

Server takes a patch request that will edit an item's duedate
### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Look at code
2. Verify sister thingy
### Links

- [TFF-154](https://msoese.atlassian.net/browse/TFF-154)
- TheFourFifths/consus-client/pull/122

### CODE REVIEW

File: src/store/student-store.js [hunk 1]
```diff
@@ -170,4 +170,15 @@ store.registerHandler('DELETE_ITEM', data => {
 store.registerHandler('DELETE_MODEL', data => {
     removeModelFromAllStudents(data.modelAddress);
 });
+
+store.registerHandler('CHANGE_ITEM_DUEDATE', data => {
+    store.waitFor(ItemStore);
+    let updatedItem = ItemStore.getItemByAddress(data.itemAddress);
+    let student = students[data.studentId];
+    for (let i = 0; i < student.items.length; i++) {
```
Comment (mattmahn) [inline src/store/student-store.js]: Nit: eww. `forEach`? `for...of`? `for...in`?

File: test/functional/edit-item-duedate.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    });
+
+    it('should notice missing itemAddress', () => {
+        let today = moment();
+        let qs = undefined;
+        let body = {
+            dueDate: today,
+            studentId: student.id
+        };
+        return patch('item/duedate', qs, body).then(() => {}).then(assert.fail).catch(e => {
```
Comment (mattmahn) [inline test/functional/edit-item-duedate.js]: Why `.then(() => {})`??

File: src/store/item-store.js [hunk 3]
```diff
@@ -56,7 +56,7 @@ function checkoutEquipment(equipment, studentId, dueDateTime) {
     equipment.forEach(equip => {
         let address = equip.address;
         let result = readAddress(address);
-        if (result.type == 'item') {
+        if (result.type === 'item') {
```
Comment (Cafferoq) [inline src/store/item-store.js]: I thought our linter was supposed to catch these

File: test/unit/store/student-store.js [hunk 4]
```diff
@@ -71,6 +73,35 @@ describe('StudentStore', () => {
...
+        }).then(() => {
+            assert.lengthOf(student.items, 2);
+            tomorrow.add(2, 'd');
+            return addAction('CHANGE_ITEM_DUEDATE', {
+                dueDate: tomorrow.format('YYYY-MM-DD'),
+                studentId: student.id,
+                itemAddress: items[0].address
+            });
+        }).then(() => {
+            tomorrow.hour(17).minute(0).second(0);
```
Comment (mattmahn) [inline test/unit/store/student-store.js]: NACK: change this to use `checkin.due_hour` and `checkin.due_minute` from config

File: test/functional/edit-item-duedate.js (hunk 2 above)
Comment (gearedmonkey) [inline test/functional/edit-item-duedate.js]: fixed.

File: test/unit/store/student-store.js (hunk 4 above)
Comment (gearedmonkey) [inline test/unit/store/student-store.js]: fixed.

**Pull request merged**